client/my-sites/plans/jetpack-plans/product-store/index.tsx

```diff
@@ -10,13 +15,40 @@ import './style.scss';
 const ProductStore: React.FC< ProductStoreProps > = ( {
 	enableUserLicensesDialog,
 	urlQueryArgs,
+	duration,
 } ) => {
 	const siteId = useSelector( getSelectedSiteId );
+	const currentPlan = useSelector( ( state ) => getSitePlan( state, siteId ) );
+	const currentPlanSlug = currentPlan?.product_slug || null;
+	const { availableProducts, purchasedProducts, includedInPlanProducts } =
+		useGetPlansGridProducts( siteId );
+
+	const allItems = useMemo(
+		() => [
+			...getProductsToDisplay( {
+				duration,
+				availableProducts,
+				purchasedProducts,
+				includedInPlanProducts,
+			} ),
+			...getPlansToDisplay( { duration, currentPlanSlug } ),
+		],
+		[ duration, availableProducts, purchasedProducts, includedInPlanProducts, currentPlanSlug ]
+	);
 
 	return (
 		<div className="jetpack-product-store">
 			{ enableUserLicensesDialog && <UserLicensesDialog siteId={ siteId } /> }
+
+			<div className="jetpack-product-store__pricing-banner">
+				<IntroPricingBanner
+					productSlugs={ allItems.map( ( { productSlug } ) => productSlug ) }
```

Comment:
Why not move this `.map()` call to `useMemo`?

Reply:
Then you could have

```ts
const productSlugs = useMemo(
	() =>
		[
			...getProductsToDisplay( {
				duration,
				availableProducts,
				purchasedProducts,
				includedInPlanProducts,
			} ),
			...getPlansToDisplay( { duration, currentPlanSlug } ),
		].map( ( { productSlug } ) => productSlug ),
	[ duration, availableProducts, purchasedProducts, includedInPlanProducts, currentPlanSlug ]
);
```

Reply:
thanks @manzoorwanijk for the suggestion. I agree this one is a lot better.. 👍 💯